### ISSUE TITLE
[Feature] map-support for extended_headers_rcpt / milter_headers

### ISSUE THREAD

**Issue by Manawyrm** [rspamd/rspamd#3400]
## Summary
The extended_headers_rcpt option in the milter_headers plugin allows for configuration of addresses that should receive a full detail info about the spam score. It would be helpful to be able to set this value as a map instead of just a list in the configuration files. 

## Motivation

To figure out why a certain (type of) mail is getting filtered, it would be handy to enable detailed spam scores in the header. We don't want to enable the detailed score for all users. 
The best solution to that would be a map (because the configuration is built and operated by a configuration managment software and written by the server administrators instead of the mail operators). 

## Describe alternatives you've considered

Setting a map as the extended_headers_rcpt doesn't seem to be supported in the plugin. I can't think of another good way to get a detailed spam score only for a few addresses. 
Using the webinterface history is not an option because of the large volume of incoming mail. 

## Target
```
# local.d/milter_headers.conf:
extended_headers_rcpt = "${DBDIR}/extended_headers_rcpt.map";
```

**Comment by vstakhov** [comment]
https://github.com/rspamd/rspamd/commit/0c90e73443254ec4684bd5a142eb627d20e7df18

**Comment by Manawyrm** [comment]
Hello Vsevolod,

thanks a lot for your commit. 
I've tested it and could not reproduce the expected behaviour. 

I looked at the code and added some debug output. 
It looks like the logic is inverted. I've changed the order of operations and inverted the if-statements: 
https://github.com/Manawyrm/rspamd/commit/c562c27e0c8a48b6838d9b08a06cdebf7959d547

The feature now works exactly as intended. 
Should I open a PR or could you pull the commit directly? 

Thanks for your work,
Tobias

**Comment by vstakhov** [comment]
> Should I open a PR or could you pull the commit directly?
Yes please.

**Comment by Manawyrm** [comment]
OK! I've created 
https://github.com/rspamd/rspamd/pull/3607 
:smiley:

**Comment by Manawyrm** [comment]
I guess this is fixed now, by #3607 and 0c90e73443254ec4684bd5a142eb627d20e7df18.

Thanks!
